### ISSUE TITLE
Error Registering LUA file

### ISSUE THREAD

**Issue by kevinmic** [aerospike/aerospike-client-java#20]
I simply want to add a value onto the end of a list.   It does not appear that I can do this via the regular api so I am trying to write a lua function that will do it for me.

Every time I try to register my lua file I get this error --

> Exception in thread "main" com.aerospike.client.AerospikeException: Registration failed: invalid_content_len
> File: null
> Line: null
> Message: null
> 	at com.aerospike.client.AerospikeClient.registerContent(AerospikeClient.java:1014)
> 	at com.aerospike.client.AerospikeClient.register(AerospikeClient.java:951)
> 	at com.alianza.aerospiketest.Main.main(Main.java:36)

Here is my example code

```java
import com.aerospike.client.AerospikeClient;
import com.aerospike.client.policy.ClientPolicy;

public class Main {
    private static final String HOST = "host";
    private static final String NAMESPACE = "namespace";

    private static AerospikeClient client;
    private static ClientPolicy policy;

    public static void main(String[] args) {
        client = new AerospikeClient(HOST, 3000);
        policy = new ClientPolicy();

        client.register(policy.readPolicyDefault, "/tmp/list_functions.lua", "list_functions.lua", com.aerospike.client.Language.LUA);
    }
}
```

lua file -- list_functions.lua
```lua
function add_to_list(record, binname, value)
  if record.MAIN == null then
    record.MAIN = list()
  end

  list.append(record.MAIN, value)
  aerospike:update(record)
end
```

**Comment by kevinmic** [comment]
I found this example here -- https://github.com/aerospike/query-with-filters/   that was doing a lua file upload.   I tried downloading it and running it and it worked fine.   I noticed that it was running version 3.0.22 so I bumped it to 3.0.35.  This broke in the same way as I described in my bug.   So I backed down to 3.0.34 and it worked just fine.

Ultimately I tracked the issue down to a change in Unpacker

This code
```java
	case 0xd1: { // signed 16 bit integer
                int val = Buffer.bytesToShort(buffer, offset);
                offset += 2;
                return getLong(val);
       }
```
was changed to
```java
			case 0xd1: { // signed 16 bit integer
				short val = Buffer.bigSigned16ToShort(buffer, offset);
				offset += 2;
				return getLong(val);
			}
```

I switched that back and released an internal version and it worked.

**Comment by kevinmic** [comment]
I have created a pull request to fix this.  Pull request #21

**Comment by BrianNichols** [comment]
We tried "https://github.com/aerospike/query-with-filters" and it runs fine under 3.0.35.  

Unpacker is only used for list and map bins.  "query-with-filters" never references list and map bins.

**Comment by wchu-citrusleaf** [comment]
In addition, correction to the lua function - in order for the updated list to be stored:
    list.append(record.MAIN, value)
should be:
    local l = record.MAIN 
    list.append(l, value)
    record.MAIN = l

That is, the record bin must be explicitly assigned, rather than passed as a parameter to the list.append() function. It's the same story for list.insert(), list.remove(), list.trim(), etc. and likewise for the similar map and byte functions.

**Comment by kevinmic** [comment]
I went and cleaned out my .m2 repository and tried again and am not able to repeat the problem.   I am not sure how I got it into that bad state in the first place as I didn't clone this repo until after I had the problems.   I will cancel that pull request.

@wchu-citrusleaf - Thanks for the tip.   I spent an hour trying to figure that out last night and finally ran across answer.   Then I went to the aerospike forums and found the answer again.   The examples I was following did not show that.   https://github.com/aerospike/udf-list-examples/blob/master/udf/campaign.lua
